api/auth_token.go

```diff
@@ -135,6 +135,26 @@ func (c *TokenAuth) RenewSelf(increment int) (*Secret, error) {
 	return ParseSecret(resp.Body)
 }
 
+// RenewTokenAsSelf behaves like renew-self, but authenticates using a provided
+// token instead of the token attached to the client.
+func (c *TokenAuth) RenewTokenAsSelf(token string, increment int) (*Secret, error) {
+	r := c.c.NewRequest("PUT", "/v1/auth/token/renew-self")
+	r.ClientToken = token
```

Comment:
Also note this is setting the token _on the request_, not the client (it's `r.ClientToken`, not `c.Token`). Since each call to NewRequest makes a new struct, I don't see this actually mutating state. Am I misinterpreting the way NewRequest works? It does not seem to modify the actual client.